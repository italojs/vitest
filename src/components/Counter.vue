<script setup>
import { ref } from 'vue'

const { initialCount } = defineProps({
  msg: String,
  initialCount: Number,
});

const methods = ref({
  increment: () => {
    console.log('increment');
    count.value++;
  },
});

const count = ref(initialCount || 0);
</script>

<template>
  <h1>{{ msg }}</h1>

  <div class="card">
    <button type="button" @click="methods.increment">The count is: {{ count }}</button>
  </div>
</template>

<style scoped>
.read-the-docs {
  color: #888;
}
</style>
