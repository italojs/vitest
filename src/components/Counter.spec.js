import { mount } from '@vue/test-utils';
import Counter from './Counter.vue'; 

describe('Counter Component', () => {
  it('increments count when button is clicked', async () => {
    const wrapper = mount(Counter, {
      props: { initialCount: 5 },
    });

    expect(wrapper.text()).toContain('The count is: 5');
    expect(wrapper.html()).toMatchSnapshot('initial state');

    const incrementSpy = vi.spyOn(wrapper.vm.methods, 'increment');
    await wrapper.find('button').trigger('click');

    expect(wrapper.text()).toContain('The count is: 6');
    expect(incrementSpy).toHaveBeenCalled();
    expect(wrapper.html()).toMatchSnapshot('after click');
  });
});
