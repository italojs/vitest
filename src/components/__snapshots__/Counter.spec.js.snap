// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Counter Component > increments count when button is clicked > after click 1`] = `
"<h1 data-v-9bd91872=""></h1>
<div data-v-9bd91872="" class="card"><button data-v-9bd91872="" type="button">The count is: 6</button></div>"
`;

exports[`Counter Component > increments count when button is clicked > initial state 1`] = `
"<h1 data-v-9bd91872=""></h1>
<div data-v-9bd91872="" class="card"><button data-v-9bd91872="" type="button">The count is: 5</button></div>"
`;
